// @medusajs/ui/build/esm/components/tooltip/tooltip.js
import * as Primitives from '@radix-ui/react-tooltip';
import * as React from 'react';
import { clx } from '../../utils/clx.js';
import { labelVariants } from '../label/label.js';

const Tooltip = ({
  children,
  content,
  open,
  defaultOpen,
  onOpenChange,
  delayDuration,
  maxWidth = 220,
  className,
  side,
  sideOffset = 4,
  onClick,
  ...props
}) => {
  return /*#__PURE__*/React.createElement(Primitives.Provider, {
    delayDuration: 100
  }, /*#__PURE__*/React.createElement(Primitives.Root, {
    open: open,
    defaultOpen: defaultOpen,
    onOpenChange: onOpenChange,
    delayDuration: delayDuration
  }, /*#__PURE__*/React.createElement(Primitives.Trigger, {
    onClick: onClick,
    asChild: true
  }, children), /*#__PURE__*/React.createElement(Primitives.Portal, null, /*#__PURE__*/React.createElement(Primitives.Content, Object.assign({
    side: side,
    sideOffset: sideOffset,
    align: "center",
    className: clx("text-ui-fg-subtle animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 bg-ui-bg-base shadow-elevation-tooltip z-[999] rounded-lg px-3 py-2", labelVariants({
      size: "xsmall",
      weight: "plus"
    }), className)
  }, props, {
    style: {
      ...props.style,
      maxWidth
    }
  }), content))));
};

export { Tooltip };
