// @medusajs/ui/build/esm/components/dropdown-menu/dropdown-menu.js
import { ChevronRightMini, CheckMini, EllipseMiniSolid } from '@medusajs/icons';
import * as Primitives from '@radix-ui/react-dropdown-menu';
import * as React from 'react';
import { clx } from '../../utils/clx.js';
import { labelVariants } from '../label/label.js';

const Root = Primitives.Root;
Root.displayName = "DropdownMenu.Root";
const Trigger = Primitives.Trigger;
Trigger.displayName = "DropdownMenu.Trigger";
const Group = Primitives.Group;
Group.displayName = "DropdownMenu.Group";
const SubMenu = Primitives.Sub;
SubMenu.displayName = "DropdownMenu.SubMenu";
const RadioGroup = Primitives.RadioGroup;
RadioGroup.displayName = "DropdownMenu.RadioGroup";
const SubMenuTrigger = /*#__PURE__*/React.forwardRef(({
  className,
  children,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.SubTrigger, Object.assign({
  ref: ref,
  className: clx("focus:bg-ui-bg-base-pressed data-[state=open]:bg-ui-bg-base-pressed flex cursor-default select-none items-center rounded-sm px-3 py-2 outline-none", labelVariants({
    size: "small"
  }), className)
}, props), children, /*#__PURE__*/React.createElement(ChevronRightMini, {
  className: "ml-auto"
})));
SubMenuTrigger.displayName = "DropdownMenu.SubMenuTrigger";
const SubMenuContent = /*#__PURE__*/React.forwardRef(({
  className,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Portal, null, /*#__PURE__*/React.createElement(Primitives.SubContent, Object.assign({
  ref: ref,
  className: clx("bg-ui-bg-base text-ui-fg-base data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 shadow-elevation-flyout z-50 min-w-[8rem] overflow-hidden rounded-lg border p-1", className)
}, props))));
SubMenuContent.displayName = "DropdownMenu.SubMenuContent";
const Content = /*#__PURE__*/React.forwardRef(({
  className,
  sideOffset = 4,
  align = "start",
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Portal, null, /*#__PURE__*/React.createElement(Primitives.Content, Object.assign({
  ref: ref,
  sideOffset: sideOffset,
  align: align,
  className: clx("bg-ui-bg-base text-ui-fg-base data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 shadow-elevation-flyout z-50 min-w-[300px] overflow-hidden rounded-lg p-1", className)
}, props))));
Content.displayName = "DropdownMenu.Content";
const Item = /*#__PURE__*/React.forwardRef(({
  className,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Item, Object.assign({
  ref: ref,
  className: clx("bg-ui-bg-base focus:bg-ui-bg-base-pressed text-ui-fg-base data-[disabled]:text-ui-fg-disabled relative flex cursor-default select-none items-center rounded-md px-3 py-2 outline-none transition-colors data-[disabled]:pointer-events-none", labelVariants({
    size: "small"
  }), className)
}, props)));
Item.displayName = "DropdownMenu.Item";
const CheckboxItem = /*#__PURE__*/React.forwardRef(({
  className,
  children,
  checked,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.CheckboxItem, Object.assign({
  ref: ref,
  className: clx("focus:bg-ui-bg-base-pressed text-ui-fg-base data-[disabled]:text-ui-fg-disabled relative flex cursor-default select-none items-center rounded-md py-2 pl-10 pr-3 text-sm outline-none transition-colors data-[disabled]:pointer-events-none", className),
  checked: checked
}, props), /*#__PURE__*/React.createElement("span", {
  className: "absolute left-3 flex h-5 w-5 items-center justify-center"
}, /*#__PURE__*/React.createElement(Primitives.ItemIndicator, null, /*#__PURE__*/React.createElement(CheckMini, null))), children));
CheckboxItem.displayName = "DropdownMenu.CheckboxItem";
const RadioItem = /*#__PURE__*/React.forwardRef(({
  className,
  children,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.RadioItem, Object.assign({
  ref: ref,
  className: clx("focus:bg-ui-bg-base-pressed hover:bg-ui-base-hover bg-ui-bg-base relative flex cursor-default select-none items-center rounded-md py-2 pl-10 pr-3 text-sm outline-none transition-colors data-[disabled]:pointer-events-none data-[state=checked]:font-medium data-[disabled]:opacity-50", labelVariants({
    size: "small"
  }), className)
}, props), /*#__PURE__*/React.createElement("span", {
  className: "absolute left-3 flex h-5 w-5 items-center justify-center"
}, /*#__PURE__*/React.createElement(Primitives.ItemIndicator, null, /*#__PURE__*/React.createElement(EllipseMiniSolid, {
  className: "text-ui-fg-base"
}))), children));
RadioItem.displayName = "DropdownMenu.RadioItem";
const Label = /*#__PURE__*/React.forwardRef(({
  className,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Label, Object.assign({
  ref: ref,
  className: clx("text-ui-fg-subtle px-2 py-1.5", labelVariants({
    size: "xsmall",
    weight: "plus"
  }), className)
}, props)));
Label.displayName = "DropdownMenu.Label";
const Separator = /*#__PURE__*/React.forwardRef(({
  className,
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Separator, Object.assign({
  ref: ref,
  className: clx("bg-ui-border-base -mx-1 my-1 h-px", className)
}, props)));
Separator.displayName = "DropdownMenu.Separator";
const Shortcut = ({
  className,
  ...props
}) => {
  return /*#__PURE__*/React.createElement("span", Object.assign({
    className: clx("text-ui-fg-subtle ml-auto tracking-widest", labelVariants({
      size: "small"
    }), className)
  }, props));
};
Shortcut.displayName = "DropdownMenu.Shortcut";
const Hint = ({
  className,
  ...props
}) => {
  return /*#__PURE__*/React.createElement("span", Object.assign({
    className: clx("text-ui-fg-subtle ml-auto tracking-widest", labelVariants({
      size: "small"
    }), className)
  }, props));
};
Hint.displayName = "DropdownMenu.Hint";
const DropdownMenu = Object.assign(Root, {
  Trigger,
  Group,
  SubMenu,
  SubMenuContent,
  SubMenuTrigger,
  Content,
  Item,
  CheckboxItem,
  RadioGroup,
  RadioItem,
  Label,
  Separator,
  Shortcut,
  Hint
});

export { DropdownMenu };
