// @medusajs/ui/build/esm/components/switch/switch.js
import * as Primitives from '@radix-ui/react-switch';
import { cva } from 'class-variance-authority';
import * as React from 'react';
import { clx } from '../../utils/clx.js';

const switchVariants = cva("bg-ui-bg-toggle-off hover:bg-ui-bg-toggle-off-hover data-[state=unchecked]:hover:after:bg-toggle-off-hover-gradient before:shadow-details-switch-background focus:shadow-details-switch-background-focus data-[state=checked]:bg-interactive disabled:!bg-ui-bg-disabled group relative inline-flex items-center rounded-full outline-none transition-all before:absolute before:inset-0 before:rounded-full before:content-[''] after:absolute after:inset-0 after:rounded-full after:content-[''] disabled:cursor-not-allowed", {
  variants: {
    size: {
      small: "h-[16px] w-[28px]",
      base: "h-[18px] w-[32px]"
    }
  },
  defaultVariants: {
    size: "base"
  }
});
const thumbVariants = cva("bg-ui-bg-base shadow-details-switch-thumb group-disabled:bg-ui-bg-disabled pointer-events-none z-10 h-[14px] w-[14px] rounded-full transition-all", {
  variants: {
    size: {
      small: "h-[12px] w-[12px] data-[state=checked]:translate-x-3.5 data-[state=unchecked]:translate-x-0.5",
      base: "h-[14px] w-[14px] data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0.5"
    }
  },
  defaultVariants: {
    size: "base"
  }
});
const Switch = /*#__PURE__*/React.forwardRef(({
  className,
  size = "base",
  ...props
}, ref) => /*#__PURE__*/React.createElement(Primitives.Root, Object.assign({
  className: clx(switchVariants({
    size
  }), className)
}, props, {
  ref: ref
}), /*#__PURE__*/React.createElement(Primitives.Thumb, {
  className: clx(thumbVariants({
    size
  }))
})));
Switch.displayName = "Switch";

export { Switch };
