// @medusajs/ui-preset/dist/index.mjs
var __getOwnPropNames = Object.getOwnPropertyNames;
var __require = /* @__PURE__ */ ((x) => typeof require !== "undefined" ? require : typeof Proxy !== "undefined" ? new Proxy(x, {
  get: (a, b) => (typeof require !== "undefined" ? require : a)[b]
}) : x)(function(x) {
  if (typeof require !== "undefined")
    return require.apply(this, arguments);
  throw Error('Dynamic require of "' + x + '" is not supported');
});
var __esm = (fn, res) => function __init() {
  return fn && (res = (0, fn[__getOwnPropNames(fn)[0]])(fn = 0)), res;
};
var __commonJS = (cb, mod) => function __require2() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};

// src/theme/extension/theme.ts
var theme;
var init_theme = __esm({
  "src/theme/extension/theme.ts"() {
    "use strict";
    theme = {
      "extend": {
        "colors": {
          "ui": {
            "code": {
              "text": {
                "base": {
                  "DEFAULT": "var(--code-text-base)"
                },
                "subtle": {
                  "DEFAULT": "var(--code-text-subtle)"
                }
              },
              "bg": {
                "base": {
                  "DEFAULT": "var(--code-bg-base)"
                },
                "header": {
                  "DEFAULT": "var(--code-bg-header)"
                }
              },
              "border": {
                "base": {
                  "DEFAULT": "var(--code-border-base)"
                },
                "strong": {
                  "DEFAULT": "var(--code-border-strong)"
                }
              },
              "icon": {
                "DEFAULT": "var(--code-icon)"
              }
            },
            "tag": {
              "green": {
                "bg": {
                  "DEFAULT": "var(--tag-green-bg)",
                  "hover": {
                    "DEFAULT": "var(--tag-green-bg-hover)"
                  }
                },
                "icon": {
                  "DEFAULT": "var(--tag-green-icon)"
                },
                "border": {
                  "DEFAULT": "var(--tag-green-border)"
                },
                "text": {
                  "DEFAULT": "var(--tag-green-text)"
                }
              },
              "neutral": {
                "bg": {
                  "DEFAULT": "var(--tag-neutral-bg)",
                  "hover": {
                    "DEFAULT": "var(--tag-neutral-bg-hover)"
                  }
                },
                "border": {
                  "DEFAULT": "var(--tag-neutral-border)"
                },
                "text": {
                  "DEFAULT": "var(--tag-neutral-text)"
                },
                "icon": {
                  "DEFAULT": "var(--tag-neutral-icon)"
                }
              },
              "red": {
                "bg": {
                  "DEFAULT": "var(--tag-red-bg)",
                  "hover": {
                    "DEFAULT": "var(--tag-red-bg-hover)"
                  }
                },
                "icon": {
                  "DEFAULT": "var(--tag-red-icon)"
                },
                "text": {
                  "DEFAULT": "var(--tag-red-text)"
                },
                "border": {
                  "DEFAULT": "var(--tag-red-border)"
                }
              },
              "orange": {
                "bg": {
                  "DEFAULT": "var(--tag-orange-bg)",
                  "hover": {
                    "DEFAULT": "var(--tag-orange-bg-hover)"
                  }
                },
                "icon": {
                  "DEFAULT": "var(--tag-orange-icon)"
                },
                "text": {
                  "DEFAULT": "var(--tag-orange-text)"
                },
                "border": {
                  "DEFAULT": "var(--tag-orange-border)"
                }
              },
              "purple": {
                "bg": {
                  "hover": {
                    "DEFAULT": "var(--tag-purple-bg-hover)"
                  },
                  "DEFAULT": "var(--tag-purple-bg)"
                },
                "text": {
                  "DEFAULT": "var(--tag-purple-text)"
                },
                "icon": {
                  "DEFAULT": "var(--tag-purple-icon)"
                },
                "border": {
                  "DEFAULT": "var(--tag-purple-border)"
                }
              },
              "blue": {
                "border": {
                  "DEFAULT": "var(--tag-blue-border)"
                },
                "bg": {
                  "DEFAULT": "var(--tag-blue-bg)",
                  "hover": {
                    "DEFAULT": "var(--tag-blue-bg-hover)"
                  }
                },
                "icon": {
                  "DEFAULT": "var(--tag-blue-icon)"
                },
                "text": {
                  "DEFAULT": "var(--tag-blue-text)"
                }
              }
            },
            "border": {
              "strong": {
                "DEFAULT": "var(--border-strong)"
              },
              "interactive": {
                "DEFAULT": "var(--border-interactive)"
              },
              "base": {
                "DEFAULT": "var(--border-base)"
              },
              "error": {
                "DEFAULT": "var(--border-error)"
              },
              "loud": {
                "DEFAULT": "var(--border-loud)",
                "transparent": {
                  "DEFAULT": "var(--border-loud-transparent)"
                },
                "muted": {
                  "DEFAULT": "var(--border-loud-muted)"
                }
              },
              "danger": {
                "DEFAULT": "var(--border-danger)"
              }
            },
            "bg": {
              "highlight": {
                "DEFAULT": "var(--bg-highlight)"
              },
              "base": {
                "hover": {
                  "DEFAULT": "var(--bg-base-hover)"
                },
                "DEFAULT": "var(--bg-base)",
                "pressed": {
                  "DEFAULT": "var(--bg-base-pressed)"
                }
              },
              "subtle": {
                "DEFAULT": "var(--bg-subtle)",
                "pressed": {
                  "DEFAULT": "var(--bg-subtle-pressed)"
                },
                "hover": {
                  "DEFAULT": "var(--bg-subtle-hover)"
                }
              },
              "component": {
                "DEFAULT": "var(--bg-component)"
              },
              "interactive": {
                "DEFAULT": "var(--bg-interactive)"
              },
              "overlay": {
                "DEFAULT": "var(--bg-overlay)"
              },
              "toggle": {
                "off": {
                  "DEFAULT": "var(--bg-toggle-off)",
                  "hover": {
                    "DEFAULT": "var(--bg-toggle-off-hover)"
                  }
                }
              },
              "disabled": {
                "DEFAULT": "var(--bg-disabled)"
              }
            },
            "fg": {
              "on": {
                "inverted": {
                  "DEFAULT": "var(--fg-on-inverted)"
                },
                "color": {
                  "DEFAULT": "var(--fg-on-color)"
                }
              },
              "interactive": {
                "hover": {
                  "DEFAULT": "var(--fg-interactive-hover)"
                },
                "DEFAULT": "var(--fg-interactive)"
              },
              "error": {
                "DEFAULT": "var(--fg-error)"
              },
              "base": {
                "DEFAULT": "var(--fg-base)"
              },
              "disabled": {
                "DEFAULT": "var(--fg-disabled)"
              },
              "subtle": {
                "DEFAULT": "var(--fg-subtle)"
              },
              "muted": {
                "DEFAULT": "var(--fg-muted)"
              }
            },
            "button": {
              "danger": {
                "DEFAULT": "var(--button-danger)",
                "pressed": {
                  "DEFAULT": "var(--button-danger-pressed)"
                },
                "hover": {
                  "DEFAULT": "var(--button-danger-hover)"
                }
              },
              "neutral": {
                "DEFAULT": "var(--button-neutral)",
                "pressed": {
                  "DEFAULT": "var(--button-neutral-pressed)"
                },
                "hover": {
                  "DEFAULT": "var(--button-neutral-hover)"
                }
              },
              "transparent": {
                "DEFAULT": "var(--button-transparent)",
                "hover": {
                  "DEFAULT": "var(--button-transparent-hover)"
                },
                "pressed": {
                  "DEFAULT": "var(--button-transparent-pressed)"
                }
              },
              "inverted": {
                "pressed": {
                  "DEFAULT": "var(--button-inverted-pressed)"
                },
                "hover": {
                  "DEFAULT": "var(--button-inverted-hover)"
                },
                "DEFAULT": "var(--button-inverted)"
              }
            }
          }
        },
        "boxShadow": {
          "details-switch-handle": "var(--details-switch-handle)",
          "details-switch-background-focus": "var(--details-switch-background-focus)",
          "details-switch-background": "var(--details-switch-background)",
          "borders-focus": "var(--borders-focus)",
          "elevation-flyout": "var(--elevation-flyout)",
          "borders-interactive": "var(--borders-interactive)",
          "buttons-secondary": "var(--buttons-secondary)",
          "buttons-danger": "var(--buttons-danger)",
          "buttons-primary-focus": "var(--buttons-primary-focus)",
          "elevation-modal": "var(--elevation-modal)",
          "borders-interactive-w-focus": "var(--borders-interactive-w-focus)",
          "borders-active": "var(--borders-active)",
          "borders-base-w-shadow": "var(--borders-base-w-shadow)",
          "borders-base": "var(--borders-base)",
          "borders-strong-w-shadow": "var(--borders-strong-w-shadow)",
          "borders-interactive-w-shadow": "var(--borders-interactive-w-shadow)",
          "buttons-primary": "var(--buttons-primary)",
          "buttons-secondary-focus": "var(--buttons-secondary-focus)",
          "buttons-danger-focus": "var(--buttons-danger-focus)",
          "elevation-tooltip": "var(--elevation-tooltip)",
          "elevation-card-hover": "var(--elevation-card-hover)",
          "elevation-card-rest": "var(--elevation-card-rest)",
          "details-contrast-on-bg-interactive": "var(--details-contrast-on-bg-interactive)",
          "borders-error": "var(--borders-error)"
        }
      }
    };
  }
});

// src/theme/tokens/colors.ts
var colors;
var init_colors = __esm({
  "src/theme/tokens/colors.ts"() {
    "use strict";
    colors = {
      "dark": {
        "--button-neutral-gradient-from": "rgba(46, 46, 50, 1)",
        "--button-neutral-gradient-to": "rgba(40, 40, 44, 1)",
        "--tag-neutral-bg-hover": "rgba(46, 46, 50, 1)",
        "--tag-blue-icon": "rgba(0, 145, 255, 1)",
        "--tag-orange-border": "rgba(87, 51, 0, 1)",
        "--button-neutral-pressed": "rgba(52, 52, 58, 1)",
        "--tag-pink-text": "rgba(246, 92, 182, 1)",
        "--tag-red-icon": "rgba(229, 72, 77, 1)",
        "--tag-transparent-border": "rgba(62, 62, 68, 1)",
        "--tag-purple-icon": "rgba(110, 86, 207, 1)",
        "--button-neutral-hover": "rgba(46, 46, 50, 1)",
        "--bg-interactive": "rgba(0, 145, 255, 1)",
        "--tag-red-bg-hover": "rgba(84, 27, 31, 1)",
        "--tag-transparent-bg": "rgba(28, 28, 31, 1)",
        "--tag-blue-bg": "rgba(16, 42, 76, 1)",
        "--tag-neutral-border": "rgba(62, 62, 68, 1)",
        "--button-danger-gradient-from": "rgba(220, 61, 67, 1)",
        "--button-danger-gradient-to": "rgba(170, 36, 41, 1)",
        "--bg-subtle-pressed": "rgba(35, 35, 38, 1)",
        "--tag-red-border": "rgba(103, 30, 34, 1)",
        "--tag-green-icon": "rgba(48, 164, 108, 1)",
        "--button-inverted-hover": "rgba(244, 242, 244, 1)",
        "--tag-red-bg": "rgba(72, 26, 29, 1)",
        "--button-danger-pressed": "rgba(229, 72, 77, 1)",
        "--bg-field-hover": "rgba(40, 40, 44, 1)",
        "--code-icon": "rgba(80, 79, 87, 1)",
        "--code-bg-base": "rgba(28, 28, 31, 1)",
        "--button-transparent-hover": "rgba(46, 46, 50, 1)",
        "--code-text-subtle": "rgba(126, 125, 134, 1)",
        "--code-border-strong": "rgba(62, 62, 68, 1)",
        "--tag-blue-text": "rgba(82, 169, 255, 1)",
        "--bg-highlight": "rgba(16, 42, 76, 1)",
        "--tag-pink-bg-hover": "rgba(80, 27, 63, 1)",
        "--border-base": "rgba(46, 46, 50, 1)",
        "--bg-component": "rgba(46, 46, 50, 1)",
        "--tag-neutral-icon": "rgba(112, 111, 120, 1)",
        "--bg-base": "rgba(28, 28, 31, 1)",
        "--button-inverted-pressed": "rgba(238, 237, 239, 1)",
        "--bg-toggle-off": "rgba(62, 62, 68, 1)",
        "--border-strong": "rgba(62, 62, 68, 1)",
        "--border-error": "rgba(229, 72, 77, 1)",
        "--button-inverted-gradient-from": "rgba(255, 255, 255, 1)",
        "--button-inverted-gradient-to": "rgba(244, 242, 244, 1)",
        "--tag-orange-text": "rgba(241, 161, 13, 1)",
        "--code-border-base": "rgba(46, 46, 50, 1)",
        "--tag-pink-icon": "rgba(214, 64, 159, 1)",
        "--tag-transparent-icon": "rgba(112, 111, 120, 1)",
        "--bg-base-hover": "rgba(35, 35, 38, 1)",
        "--bg-subtle-hover": "rgba(28, 28, 31, 1)",
        "--bg-subtle": "rgba(22, 22, 24, 1)",
        "--tag-purple-text": "rgba(158, 140, 252, 1)",
        "--tag-transparent-text": "rgba(160, 159, 166, 1)",
        "--tag-pink-bg": "rgba(69, 26, 55, 1)",
        "--tag-transparent-bg-hover": "rgba(35, 35, 38, 1)",
        "--bg-overlay": "rgba(22, 22, 24, 0.7)",
        "--tag-green-text": "rgba(76, 195, 138, 1)",
        "--bg-field": "rgba(35, 35, 38, 1)",
        "--button-disabled": "rgba(40, 40, 44, 1)",
        "--tag-blue-bg-hover": "rgba(15, 48, 88, 1)",
        "--button-danger-hover": "rgba(170, 36, 41, 1)",
        "--code-bg-header": "rgba(22, 22, 24, 1)",
        "--bg-base-pressed": "rgba(40, 40, 44, 1)",
        "--tag-orange-bg-hover": "rgba(74, 41, 0, 1)",
        "--button-transparent": "rgba(0, 0, 0, 0)",
        "--button-transparent-pressed": "rgba(46, 46, 50, 1)",
        "--tag-pink-border": "rgba(96, 29, 72, 1)",
        "--tag-neutral-bg": "rgba(40, 40, 44, 1)",
        "--code-text-base": "rgba(237, 237, 239, 1)",
        "--tag-green-border": "rgba(22, 68, 48, 1)",
        "--tag-red-text": "rgba(255, 99, 105, 1)",
        "--tag-green-bg-hover": "rgba(19, 57, 41, 1)",
        "--tag-purple-bg-hover": "rgba(50, 39, 95, 1)",
        "--tag-orange-icon": "rgba(255, 178, 36, 1)",
        "--tag-purple-bg": "rgba(44, 34, 80, 1)",
        "--tag-green-bg": "rgba(17, 49, 35, 1)",
        "--tag-blue-border": "rgba(13, 56, 104, 1)",
        "--tag-neutral-text": "rgba(160, 159, 166, 1)",
        "--tag-purple-border": "rgba(57, 44, 114, 1)",
        "--tag-orange-bg": "rgba(63, 34, 0, 1)",
        "--border-interactive": "rgba(0, 145, 255, 1)",
        "--border-loud": "rgba(237, 237, 239, 1)",
        "--fg-on-color": "rgba(255, 255, 255, 1)",
        "--fg-interactive-hover": "rgba(54, 158, 255, 1)",
        "--fg-error": "rgba(229, 72, 77, 1)",
        "--fg-subtle": "rgba(126, 125, 134, 1)",
        "--fg-base": "rgba(237, 237, 239, 1)",
        "--fg-disabled": "rgba(80, 79, 87, 1)",
        "--fg-muted": "rgba(112, 111, 120, 1)",
        "--fg-interactive": "rgba(0, 145, 255, 1)",
        "--fg-on-inverted": "rgba(26, 21, 35, 1)"
      },
      "light": {
        "--code-text-base": "rgba(249, 250, 251, 1)",
        "--tag-green-bg": "rgba(209, 250, 229, 1)",
        "--border-strong": "rgba(209, 213, 219, 1)",
        "--border-interactive": "rgba(59, 130, 246, 1)",
        "--bg-highlight": "rgba(239, 246, 255, 1)",
        "--bg-base-hover": "rgba(249, 250, 251, 1)",
        "--tag-neutral-bg": "rgba(243, 244, 246, 1)",
        "--tag-red-bg": "rgba(255, 228, 230, 1)",
        "--tag-orange-bg": "rgba(254, 244, 199, 1)",
        "--bg-base": "rgba(255, 255, 255, 1)",
        "--border-base": "rgba(229, 231, 235, 1)",
        "--tag-green-icon": "rgba(5, 150, 105, 1)",
        "--code-bg-base": "rgba(3, 7, 18, 1)",
        "--tag-purple-bg-hover": "rgba(221, 214, 254, 1)",
        "--tag-blue-border": "rgba(191, 219, 254, 1)",
        "--tag-orange-icon": "rgba(217, 119, 6, 1)",
        "--tag-purple-bg": "rgba(237, 233, 254, 1)",
        "--code-bg-header": "rgba(17, 24, 39, 1)",
        "--bg-subtle": "rgba(249, 250, 251, 1)",
        "--tag-purple-text": "rgba(109, 40, 217, 1)",
        "--tag-blue-bg": "rgba(219, 234, 254, 1)",
        "--tag-blue-icon": "rgba(37, 99, 235, 1)",
        "--code-border-base": "rgba(31, 41, 55, 1)",
        "--code-border-strong": "rgba(55, 65, 81, 1)",
        "--bg-component": "rgba(241, 243, 245, 1)",
        "--code-text-subtle": "rgba(107, 114, 128, 1)",
        "--code-icon": "rgba(75, 85, 99, 1)",
        "--border-error": "rgba(225, 29, 72, 1)",
        "--border-loud": "rgba(3, 7, 18, 1)",
        "--fg-on-inverted": "rgba(255, 255, 255, 1)",
        "--fg-on-color": "rgba(255, 255, 255, 1)",
        "--button-danger": "rgba(225, 29, 72, 1)",
        "--button-danger-gradient-from": "rgba(255, 255, 255, 1)",
        "--button-danger-gradient-to": "rgba(255, 255, 255, 0)",
        "--button-neutral": "rgba(255, 255, 255, 1)",
        "--button-neutral-gradient-from": "rgba(3, 7, 18, 0)",
        "--button-neutral-gradient-to": "rgba(3, 7, 18, 1)",
        "--button-danger-pressed": "rgba(225, 29, 72, 1)",
        "--button-danger-pressed-gradient-from": "rgba(255, 255, 255, 0)",
        "--button-danger-pressed-gradient-to": "rgba(255, 255, 255, 1)",
        "--button-transparent": "rgba(255, 255, 255, 0)",
        "--button-inverted-pressed": "rgba(17, 24, 39, 1)",
        "--button-inverted-pressed-gradient-from": "rgba(255, 255, 255, 0)",
        "--button-inverted-pressed-gradient-to": "rgba(255, 255, 255, 1)",
        "--button-inverted-hover": "rgba(17, 24, 39, 1)",
        "--button-inverted-hover-gradient-from": "rgba(255, 255, 255, 1)",
        "--button-inverted-hover-gradient-to": "rgba(255, 255, 255, 0)",
        "--button-inverted": "rgba(17, 24, 39, 1)",
        "--button-inverted-gradient-from": "rgba(255, 255, 255, 1)",
        "--button-inverted-gradient-to": "rgba(255, 255, 255, 0)",
        "--button-danger-hover": "rgba(225, 29, 72, 1)",
        "--button-danger-hover-gradient-from": "rgba(255, 255, 255, 1)",
        "--button-danger-hover-gradient-to": "rgba(255, 255, 255, 0)",
        "--bg-interactive": "rgba(59, 130, 246, 1)",
        "--fg-interactive-hover": "rgba(37, 99, 235, 1)",
        "--fg-interactive": "rgba(59, 130, 246, 1)",
        "--fg-error": "rgba(225, 29, 72, 1)",
        "--border-loud-transparent": "rgba(3, 7, 18, 0)",
        "--border-danger": "rgba(190, 18, 60, 1)",
        "--border-loud-muted": "rgba(3, 7, 18, 0.1)",
        "--bg-overlay": "rgba(3, 7, 18, 0.4)",
        "--bg-toggle-off": "rgba(229, 231, 235, 1)",
        "--fg-base": "rgba(3, 7, 18, 1)",
        "--fg-disabled": "rgba(209, 213, 219, 1)",
        "--bg-subtle-pressed": "rgba(229, 231, 235, 1)",
        "--fg-subtle": "rgba(75, 85, 99, 1)",
        "--fg-muted": "rgba(156, 163, 175, 1)",
        "--bg-base-pressed": "rgba(243, 244, 246, 1)",
        "--bg-subtle-hover": "rgba(243, 244, 246, 1)",
        "--tag-neutral-border": "rgba(229, 231, 235, 1)",
        "--tag-green-bg-hover": "rgba(167, 243, 208, 1)",
        "--tag-blue-bg-hover": "rgba(191, 219, 254, 1)",
        "--tag-red-icon": "rgba(225, 29, 72, 1)",
        "--tag-neutral-text": "rgba(75, 85, 99, 1)",
        "--tag-red-bg-hover": "rgba(254, 205, 211, 1)",
        "--tag-red-text": "rgba(190, 18, 60, 1)",
        "--tag-purple-icon": "rgba(124, 58, 237, 1)",
        "--tag-blue-text": "rgba(29, 78, 216, 1)",
        "--tag-orange-bg-hover": "rgba(253, 230, 138, 1)",
        "--tag-neutral-bg-hover": "rgba(229, 231, 235, 1)",
        "--tag-purple-border": "rgba(221, 214, 254, 1)",
        "--tag-orange-text": "rgba(180, 83, 9, 1)",
        "--tag-neutral-icon": "rgba(107, 114, 128, 1)",
        "--tag-orange-border": "rgba(253, 230, 138, 1)",
        "--tag-red-border": "rgba(254, 205, 211, 1)",
        "--tag-green-border": "rgba(167, 243, 208, 1)",
        "--tag-green-text": "rgba(4, 120, 87, 1)",
        "--bg-disabled": "rgba(243, 244, 246, 1)",
        "--button-transparent-hover": "rgba(249, 250, 251, 1)",
        "--button-neutral-pressed": "rgba(255, 255, 255, 1)",
        "--button-neutral-pressed-gradient-from": "rgba(3, 7, 18, 1)",
        "--button-neutral-pressed-gradient-to": "rgba(3, 7, 18, 0)",
        "--button-transparent-pressed": "rgba(243, 244, 246, 1)",
        "--button-neutral-hover": "rgba(255, 255, 255, 1)",
        "--button-neutral-hover-gradient-from": "rgba(3, 7, 18, 0)",
        "--button-neutral-hover-gradient-to": "rgba(3, 7, 18, 1)",
        "--bg-toggle-off-hover": "rgba(229, 231, 235, 1)",
        "--bg-toggle-off-hover-gradient-from": "rgba(3, 7, 18, 0)",
        "--bg-toggle-off-hover-gradient-to": "rgba(3, 7, 18, 1)"
      }
    };
  }
});

// src/theme/tokens/components.ts
var components;
var init_components = __esm({
  "src/theme/tokens/components.ts"() {
    "use strict";
    components = {
      ".button-neutral-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-neutral-gradient-from), var(--button-neutral-gradient-to))",
        "opacity": "2%"
      },
      ".button-danger-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-danger-gradient-from), var(--button-danger-gradient-to))",
        "opacity": "12%"
      },
      ".button-inverted-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-inverted-gradient-from), var(--button-inverted-gradient-to))",
        "opacity": "12%"
      },
      ".button-danger-pressed-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-danger-pressed-gradient-from), var(--button-danger-pressed-gradient-to))",
        "opacity": "16%"
      },
      ".button-inverted-pressed-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-inverted-pressed-gradient-from), var(--button-inverted-pressed-gradient-to))",
        "opacity": "16%"
      },
      ".button-inverted-hover-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-inverted-hover-gradient-from), var(--button-inverted-hover-gradient-to))",
        "opacity": "16%"
      },
      ".button-danger-hover-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-danger-hover-gradient-from), var(--button-danger-hover-gradient-to))",
        "opacity": "16%"
      },
      ".button-neutral-pressed-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-neutral-pressed-gradient-from), var(--button-neutral-pressed-gradient-to))",
        "opacity": "6%"
      },
      ".button-neutral-hover-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--button-neutral-hover-gradient-from), var(--button-neutral-hover-gradient-to))",
        "opacity": "6%"
      },
      ".bg-toggle-off-hover-gradient": {
        "backgroundImage": "linear-gradient(0deg, var(--bg-toggle-off-hover-gradient-from), var(--bg-toggle-off-hover-gradient-to))",
        "opacity": "6%"
      }
    };
  }
});

// src/theme/tokens/effects.ts
var effects;
var init_effects = __esm({
  "src/theme/tokens/effects.ts"() {
    "use strict";
    effects = {
      "dark": {
        "--elevation-flyout": "0px 0px 0px 1px rgba(255, 255, 255, 0.1), 0px 8px 16px 0px rgba(0, 0, 0, 0.32)",
        "--form-focus": "0px 0px 0px 2px rgba(28, 28, 31, 1), 0px 0px 0px 4px rgba(0, 145, 255, 1)",
        "--elevation-modal": "0px 0px 0px 1px rgba(255, 255, 255, 0.1), 0px 16px 32px 0px rgba(0, 0, 0, 0.32), 0px 2px 24px 0px rgba(0, 0, 0, 0.32)",
        "--elevation-card-hover": "0px 0px 0px 1px rgba(255, 255, 255, 0.1), 0px 1px 2px -1px rgba(255, 255, 255, 0.16), 0px 2px 8px 0px rgba(0, 0, 0, 0.4)",
        "--elevation-card-rest": "0px 0px 0px 1px rgba(255, 255, 255, 0.1), 0px 1px 2px -1px rgba(255, 255, 255, 0.16), 0px 2px 4px 0px rgba(0, 0, 0, 0.4)",
        "--form-active": "0px 0px 0px 3px rgba(16, 42, 76, 1)",
        "--elevation-tooltip": "0px 0px 0px 1px rgba(255, 255, 255, 0.1), 0px 4px 8px 0px rgba(0, 0, 0, 0.32)"
      },
      "light": {
        "--details-switch-handle": "0px 0px 2px 1px rgba(255, 255, 255, 1) inset, 0px 1px 0px 0px rgba(255, 255, 255, 1) inset, 0px 0px 0px 0.5px rgba(3, 7, 18, 0.02), 0px 5px 4px 0px rgba(3, 7, 18, 0.02), 0px 3px 3px 0px rgba(3, 7, 18, 0.04), 0px 1px 2px 0px rgba(3, 7, 18, 0.12), 0px 0px 1px 0px rgba(3, 7, 18, 0.08)",
        "--details-switch-background-focus": "0px 0px 0px 1px rgba(255, 255, 255, 1), 0px 0px 0px 3px rgba(59, 130, 246, 0.6), 0px 1px 1px 0px rgba(3, 7, 18, 0.04) inset, 0px 2px 4px 0px rgba(3, 7, 18, 0.04) inset, 0px 0px 0px 0.5px rgba(3, 7, 18, 0.06) inset, 0px 0px 8px 0px rgba(3, 7, 18, 0.02) inset, 0px 1px 0px 0px rgba(255, 255, 255, 1), 0px 2px 4px 0px rgba(3, 7, 18, 0.04)",
        "--details-switch-background": "0px 1px 1px 0px rgba(3, 7, 18, 0.04) inset, 0px 2px 4px 0px rgba(3, 7, 18, 0.04) inset, 0px 0px 0px 0.5px rgba(3, 7, 18, 0.06) inset, 0px 0px 8px 0px rgba(3, 7, 18, 0.02) inset, 0px 1px 0px 0px rgba(255, 255, 255, 1), 0px 2px 4px 0px rgba(3, 7, 18, 0.04)",
        "--borders-focus": "0px 0px 0px 1px rgba(255, 255, 255, 1), 0px 0px 0px 3px rgba(59, 130, 246, 0.6)",
        "--elevation-flyout": "0px 0px 0px 1px rgba(3, 7, 18, 0.08), 0px 8px 16px 0px rgba(3, 7, 18, 0.08)",
        "--borders-interactive": "0px 0px 0px 1px rgba(59, 130, 246, 1)",
        "--buttons-secondary": "0px -1px 0px 0px rgba(3, 7, 18, 0.08) inset, 0px 1px 1px 0px rgba(3, 7, 18, 0.06)",
        "--buttons-danger": "0px 1px 0px 0px rgba(255, 255, 255, 0.16) inset, 0px 1px 1px 0px rgba(190, 18, 60, 0.2)",
        "--buttons-primary-focus": "0px 1px 0px 0px rgba(255, 255, 255, 0.16) inset, 0px 1px 1px 0px rgba(3, 7, 18, 0.2), 0px 0px 0px 1px rgba(255, 255, 255, 1), 0px 0px 0px 3px rgba(59, 130, 246, 0.6)",
        "--elevation-modal": "0px 0px 0px 1px rgba(255, 255, 255, 1) inset, 0px 0px 0px 2px rgba(229, 231, 235, 0.4) inset, 0px 0px 0px 1px rgba(3, 7, 18, 0.08), 0px 16px 32px 0px rgba(3, 7, 18, 0.08), 0px 2px 24px 0px rgba(3, 7, 18, 0.08)",
        "--borders-interactive-w-focus": "0px 1px 2px 0px rgba(30, 58, 138, 0.5), 0px 0px 0px 1px rgba(59, 130, 246, 1), 0px 0px 0px 2px rgba(255, 255, 255, 1), 0px 0px 0px 4px rgba(59, 130, 246, 0.6)",
        "--borders-active": "0px 0px 0px 3px rgba(59, 130, 246, 0.2)",
        "--borders-base-w-shadow": "0px 1px 2px 0px rgba(3, 7, 18, 0.2), 0px 0px 0px 1px rgba(229, 231, 235, 1)",
        "--borders-base": "0px 0px 0px 1px rgba(229, 231, 235, 1)",
        "--borders-strong-w-shadow": "0px 1px 2px 0px rgba(3, 7, 18, 0.3), 0px 0px 0px 1px rgba(209, 213, 219, 1)",
        "--borders-interactive-w-shadow": "0px 1px 2px 0px rgba(30, 58, 138, 0.5), 0px 0px 0px 1px rgba(59, 130, 246, 1)",
        "--buttons-primary": "0px 1px 0px 0px rgba(255, 255, 255, 0.16) inset, 0px 1px 1px 0px rgba(3, 7, 18, 0.2)",
        "--buttons-secondary-focus": "0px -1px 0px 0px rgba(3, 7, 18, 0.08) inset, 0px 1px 1px 0px rgba(3, 7, 18, 0.06), 0px 0px 0px 1px rgba(255, 255, 255, 1), 0px 0px 0px 3px rgba(59, 130, 246, 0.6)",
        "--buttons-danger-focus": "0px 1px 0px 0px rgba(255, 255, 255, 0.16) inset, 0px 1px 1px 0px rgba(190, 18, 60, 0.2), 0px 0px 0px 1px rgba(255, 255, 255, 1), 0px 0px 0px 3px rgba(59, 130, 246, 0.6)",
        "--elevation-tooltip": "0px 0px 0px 1px rgba(3, 7, 18, 0.08), 0px 4px 8px 0px rgba(3, 7, 18, 0.08)",
        "--elevation-card-hover": "0px 0px 0px 1px rgba(3, 7, 18, 0.08), 0px 1px 2px -1px rgba(3, 7, 18, 0.08), 0px 2px 8px 0px rgba(3, 7, 18, 0.1)",
        "--elevation-card-rest": "0px 0px 0px 1px rgba(3, 7, 18, 0.08), 0px 1px 2px -1px rgba(3, 7, 18, 0.08), 0px 2px 4px 0px rgba(3, 7, 18, 0.04)",
        "--details-contrast-on-bg-interactive": "0px 1px 2px 0px rgba(30, 58, 138, 0.6)",
        "--borders-error": "0px 0px 0px 3px rgba(225, 29, 72, 0.15)"
      }
    };
  }
});

// src/plugin.ts
import { fontFamily } from "tailwindcss/defaultTheme";
import plugin from "tailwindcss/plugin";
var plugin_default;
var init_plugin = __esm({
  "src/plugin.ts"() {
    "use strict";
    init_theme();
    init_colors();
    init_components();
    init_effects();
    plugin_default = plugin(
      function medusaUi({ addBase, addComponents, addUtilities, config, theme: theme2 }) {
        const [darkMode, className = ".dark"] = [].concat(
          config("darkMode", "media")
        );
        addBase({
          body: {
            color: config("theme.colors.ui.fg.base")
          },
          "*": {
            borderColor: theme2("colors.ui.border.base")
          }
        });
        addComponents(components);
        addBase({
          ":root": { ...colors.light, ...effects.light }
        });
        if (darkMode === "class") {
          addBase({
            [className]: { ...colors.dark, ...effects.dark }
          });
        } else {
          addBase({
            "@media (prefers-color-scheme: dark)": {
              ":root": { ...colors.dark, ...effects.dark }
            }
          });
        }
      },
      {
        theme: {
          extend: {
            ...theme.extend,
            fontFamily: {
              sans: ["Inter", ...fontFamily.sans],
              mono: ["Roboto Mono", ...fontFamily.mono]
            }
          }
        }
      }
    );
  }
});

// src/preset.ts
var preset;
var init_preset = __esm({
  "src/preset.ts"() {
    "use strict";
    init_plugin();
    preset = {
      content: [],
      plugins: [plugin_default, __require("tailwindcss-animate")]
    };
  }
});

// src/index.ts
var require_src = __commonJS({
  "src/index.ts"(exports, module) {
    init_preset();
    module.exports = preset;
  }
});
export default require_src();
