// @medusajs/ui/build/esm/hooks/use-alert-dialog/dialog.js
import * as React from 'react';
import { clx } from '../../utils/clx.js';
import { AlertDialog } from '../../components/alert-dialog/alert-dialog.js';
import { Label, labelVariants } from '../../components/label/label.js';
import { Input } from '../../components/input/input.js';

const Dialog = ({
  open,
  title,
  description,
  verificationText,
  cancelText = "Cancel",
  confirmText = "Confirm",
  onConfirm,
  onCancel
}) => {
  const [userInput, setUserInput] = React.useState("");
  const handleUserInput = event => {
    setUserInput(event.target.value);
  };
  const validInput = React.useMemo(() => {
    if (!verificationText) {
      return true;
    }
    return userInput === verificationText;
  }, [userInput, verificationText]);
  const handleFormSubmit = event => {
    event.preventDefault();
    if (validInput) {
      onConfirm();
    }
  };
  React.useEffect(() => {
    const handleEscape = event => {
      if (event.key === "Escape" && open) {
        onCancel();
      }
    };
    document.addEventListener("keydown", handleEscape);
    return () => {
      document.removeEventListener("keydown", handleEscape);
    };
  }, [onCancel, open]);
  return /*#__PURE__*/React.createElement(AlertDialog, {
    open: open
  }, /*#__PURE__*/React.createElement(AlertDialog.Content, null, /*#__PURE__*/React.createElement(AlertDialog.Header, null, /*#__PURE__*/React.createElement(AlertDialog.Title, null, title), /*#__PURE__*/React.createElement(AlertDialog.Description, null, description)), verificationText && /*#__PURE__*/React.createElement("form", {
    onSubmit: handleFormSubmit
  }, /*#__PURE__*/React.createElement("fieldset", {
    className: "border-ui-border-base mt-6 flex flex-col gap-y-4 border-y p-6"
  }, /*#__PURE__*/React.createElement(Label, {
    htmlFor: "verificationText",
    className: "text-subtle"
  }, "Please type", " ", /*#__PURE__*/React.createElement("span", {
    className: clx(labelVariants({
      size: "base",
      weight: "plus"
    }), "text-ui-fg-base")
  }, verificationText), " ", "to confirm."), /*#__PURE__*/React.createElement(Input, {
    autoFocus: true,
    autoComplete: "off",
    id: "verificationText",
    placeholder: verificationText,
    onChange: handleUserInput
  }))), /*#__PURE__*/React.createElement(AlertDialog.Footer, null, /*#__PURE__*/React.createElement(AlertDialog.Cancel, {
    onClick: onCancel
  }, cancelText), /*#__PURE__*/React.createElement(AlertDialog.Action, {
    onClick: verificationText ? undefined : onConfirm,
    disabled: !validInput,
    type: verificationText ? "submit" : "button"
  }, confirmText))));
};
var Dialog$1 = Dialog;

export { Dialog$1 as default };
