// @medusajs/ui/build/cjs/components/alert-dialog/alert-dialog.js
'use strict';

var Primitives = require('@radix-ui/react-alert-dialog');
var React = require('react');
var clx = require('../../utils/clx.js');
var heading = require('../heading/heading.js');
var button = require('../button/button.js');
var label = require('../label/label.js');

function _interopNamespaceDefault(e) {
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var Primitives__namespace = /*#__PURE__*/_interopNamespaceDefault(Primitives);
var React__namespace = /*#__PURE__*/_interopNamespaceDefault(React);

const Root = Primitives__namespace.Root;
Root.displayName = "AlertDialog.Root";
const Trigger = Primitives__namespace.Trigger;
Trigger.displayName = "AlertDialog.Trigger";
const Portal = ({
  className,
  ...props
}) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.AlertDialogPortal, Object.assign({
    className: clx.clx(className)
  }, props));
};
Portal.displayName = "AlertDialog.Portal";
const Overlay = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Overlay, Object.assign({
    ref: ref,
    className: clx.clx("bg-ui-bg-overlay fixed inset-0 z-50 backdrop-blur-sm",
    // "data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0", // Re-enable when Admin UI has been cleaned up
    className)
  }, props));
});
Overlay.displayName = "AlertDialog.Overlay";
const Title = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Title, Object.assign({
    ref: ref,
    className: clx.clx(className)
  }, props, {
    asChild: true
  }), /*#__PURE__*/React__namespace.createElement(heading.Heading, {
    level: "h2"
  }, children));
});
Title.displayName = "AlertDialog.Title";
const Content = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Portal, null, /*#__PURE__*/React__namespace.createElement(Overlay, null), /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Content, Object.assign({
    ref: ref,
    className: clx.clx("bg-ui-bg-base shadow-elevation-flyout fixed left-[50%] top-[50%] z-50 flex w-full max-w-[400px] translate-x-[-50%] translate-y-[-50%] flex-col rounded-lg border focus:outline-none",
    // "data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] duration-200",  // Re-enable when Admin UI has been cleaned up
    className)
  }, props)));
});
Content.displayName = "AlertDialog.Content";
const Description = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Description, Object.assign({
    ref: ref,
    className: clx.clx("text-subtle", label.labelVariants({
      size: "base"
    }), className)
  }, props));
});
Description.displayName = "AlertDialog.Description";
const Action = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Action, Object.assign({
    ref: ref,
    className: clx.clx("", className)
  }, props, {
    asChild: true
  }), /*#__PURE__*/React__namespace.createElement(button.Button, {
    variant: "danger"
  }, children));
});
Action.displayName = "AlertDialog.Action";
const Cancel = /*#__PURE__*/React__namespace.forwardRef(({
  className,
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/React__namespace.createElement(Primitives__namespace.Cancel, Object.assign({
    ref: ref,
    className: clx.clx(className)
  }, props, {
    asChild: true
  }), /*#__PURE__*/React__namespace.createElement(button.Button, {
    variant: "secondary"
  }, children));
});
Cancel.displayName = "AlertDialog.Cancel";
const Header = ({
  className,
  ...props
}) => {
  return /*#__PURE__*/React__namespace.createElement("div", Object.assign({
    className: clx.clx("flex flex-col gap-y-1 px-6 pt-6", className)
  }, props));
};
const Footer = ({
  className,
  ...props
}) => {
  return /*#__PURE__*/React__namespace.createElement("div", Object.assign({
    className: clx.clx("flex items-center justify-end gap-x-2 p-6", className)
  }, props));
};
const AlertDialog = Object.assign(Root, {
  Trigger,
  Content,
  Title,
  Description,
  Action,
  Cancel,
  Header,
  Footer
});

exports.AlertDialog = AlertDialog;
