// @medusajs/ui/build/cjs/components/heading/heading.js
'use strict';

var classVarianceAuthority = require('class-variance-authority');
var React = require('react');
var clx = require('../../utils/clx.js');

function _interopNamespaceDefault(e) {
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespaceDefault(React);

const headingVariants = classVarianceAuthority.cva("font-sans font-medium", {
  variants: {
    level: {
      h1: "text-xl leading-8",
      h2: "text-base leading-6",
      h3: "text-sm leading-5"
    }
  },
  defaultVariants: {
    level: "h1"
  }
});
const Heading = ({
  level,
  className,
  ...props
}) => {
  const Component = level ? level : "h1";
  return /*#__PURE__*/React__namespace.createElement(Component, Object.assign({
    className: clx.clx(headingVariants({
      level
    }), className)
  }, props));
};

exports.Heading = Heading;
exports.headingVariants = headingVariants;
